In [2]:
import pandas as pd
import numpy as np

df = pd.read_csv("./data/train.csv")
df.shape

(16947, 52)

In [3]:
# columnas con valor unico -> no aportan informacion
nunique = df.apply(pd.Series.nunique)
cols_to_drop = nunique[nunique == 1].index
cols_to_drop

Index(['Submitted_for_Approval', 'Last_Activity', 'ASP_(converted)_Currency',
       'Actual_Delivery_Date', 'Prod_Category_A'],
      dtype='object')

In [4]:
# Elimino columnas que no aporta informacion
df.drop(cols_to_drop, axis=1, inplace = True)
df.shape

(16947, 47)

In [5]:
# Las propuestas diferentes a Closed Won o Closed Lost representan una infima cantidad (64/16mil)
# Son propuestas en curso que estan todavia en diferententes fases y no aportan informacion para predecir
# si la venta se produce o no
# son la moyoria del 2019 o 2020, y una que quedo colgada del 2017
df[(df['Stage']!= 'Closed Lost') & (df['Stage']!= 'Closed Won') ].shape

(64, 47)

In [6]:
# elimino esas 64 columnas
df = df[(df['Stage']== 'Closed Lost') | (df['Stage']== 'Closed Won') ]

In [12]:
df = df[df.Total_Amount.notna()]
df.Price.value_counts()

None     15859
Other      609
0.24        27
0.41        22
0.28        20
0.27        18
0.29        17
0.32        15
0.38        14
0.35        14
0.23        13
0.33        12
0.42        12
0.3         11
0.26        11
0.34        11
0.31        10
0.245        8
0.305        7
0.25         7
0.39         6
0.45         6
0.335        6
0.47         6
0.225        6
0.325        6
0.22         6
0.2          5
0.36         4
0.285        4
0.275        4
0.37         4
0.445        4
0.355        3
0.265        3
0.255        3
0.295        3
0.56         2
0.4          2
0.215        2
0.505        2
0.49         2
0.235        2
0.44         2
0.345        2
0.375        2
0.315        2
0.48         2
0.52         2
0.465        1
0.365        1
0.21         1
0.55         1
Name: Price, dtype: int64

Como se puede ver, en el 15859 de 16824 registros el valor es None, si contamos tambien los que tiene Other estamos hablando de 16468/16824.
Por lo que podriamos no utilizar

In [ ]:
df.Size.value_counts()

En este caso, podemos ver que la columna Size, tiene informacion sonbre apenas el 3.47%(586/16888b) de los datos, con lo que tampoco la tendremos en cuenta.

In [15]:
df.drop('Size',
  axis='columns', inplace=True)
df.drop('Price',
  axis='columns', inplace=True)

In [7]:
df.to_csv(r'./data/train_clean.csv', index = False)
df_clean = pd.read_csv("./data/train_clean.csv")
df_clean.shape

(16883, 47)